staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -666,6 +745,35 @@ func (p *sharedProcessor) resyncCheckPeriodChanged(resyncCheckPeriod time.Durati
 	}
 }
 
+func (p *sharedProcessor) removeListenerFor(handler ResourceEventHandler) {
+	p.listenersLock.Lock()
+	defer p.listenersLock.Unlock()
+
+	listener := p.removeListenerLockedFor(handler)
+	if p.listenersStarted && listener != nil {
+		close(listener.addCh)
+	}
+}
+
+func (p *sharedProcessor) removeListenerLockedFor(handler ResourceEventHandler) *processorListener {
+	if !reflect.ValueOf(handler).Type().Comparable() {
+		return nil
+	}
+	var listener *processorListener
+	for i, l := range p.listeners {
+		if reflect.ValueOf(l.handler).Type().Comparable() && l.handler == handler {
+			listener = l
+			p.listeners = append(p.listeners[:i], p.listeners[i+1:]...)
```

Comment:
I think you will get a `slice bounds out of range` panic here because you are modifying the size of slice as you iterate over it (e.g if the handler to be removed exists at any spot in the slice and at the end).

Reply:
You are right, if there would be multiple matches. To be safe I'll change this to an index based loop

Reply:
There are even more issues, if there are multiple registrations for the same handler. All the listeners must be stopped, not only the first/last one.
I fixed this accordingly.